pkg/util/deployment/deployment.go

```diff
@@ -142,6 +144,136 @@ func GetNewReplicaSetFromList(deployment extensions.Deployment, c clientset.Inte
 	return nil, nil
 }
 
+// rsListWithHashKeySynced returns a list of rs the deployment targets, with pod-template-hash information synced.
+func rsListWithHashKeySynced(deployment extensions.Deployment, c clientset.Interface, getRSList rsListFunc, getPodList podListFunc) ([]extensions.ReplicaSet, error) {
+	namespace := deployment.Namespace
+	selector, err := unversioned.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
+	options := api.ListOptions{LabelSelector: selector}
+	rsList, err := getRSList(namespace, options)
+	if err != nil {
+		return nil, err
+	}
+	syncedRSList := []extensions.ReplicaSet{}
+	for _, rs := range rsList {
+		// Add pod-template-hash information if it's not in the rs
+		if !labelsutil.SelectorHasLabel(rs.Spec.Selector, extensions.DefaultDeploymentUniqueLabelKey) {
+			updatedRS, err := addHashKeyToReplicaSet(deployment, c, rs, getPodList)
+			if err != nil {
+				return nil, err
+			}
+			syncedRSList = append(syncedRSList, *updatedRS)
+		}
+		syncedRSList = append(syncedRSList, rs)
```

Comment:
@janetkuo missed a continue in the end of the if above but since that check is redundant, we will append only once.